src/components/data/DataTable.js

```diff
@@ -349,6 +353,15 @@ const DataTable = props => {
       },
       onDismiss: () => setUpdatingEntity(undefined)
     }),
+    !!deletingColumn && h(DeleteEntityColumnModal, {
+      workspaceId: { namespace, name },
+      column: deletingColumn,
+      onSuccess: () => {
+        setDeletingColumn(undefined)
+        loadMetadata()
```

Comment:
the call to `loadMetadata()` here, because it occurs directly after deleting the column, is extremely unlikely to be cached in the back end. For workspaces with lots of rows/columns, this could cause a significant delay. Since the operation of deleting a column is targeted and well-known, would it be possible to inspect the current `entityMetadata` value, delete the column from that value in-memory and set it back into state, without ever needing to make this expensive ajax call? This feels important to try to optimize.